components/brave_wallet/browser/zcash/zcash_rpc.cc

```diff
@@ -419,6 +484,56 @@ void ZCashRpc::GetTransaction(const std::string& chain_id,
       kMaxBodySize);
 }
 
+void ZCashRpc::GetCompactBlocks(const std::string& chain_id,
+                                uint32_t from,
+                                uint32_t to,
+                                GetCompactBlocksCallback callback) {
+  GURL request_url = MakeGetCompactBlocksURL(GetNetworkURL(chain_id));
+
+  if (!request_url.is_valid()) {
+    std::move(callback).Run(
+        base::unexpected(l10n_util::GetStringUTF8(IDS_WALLET_INTERNAL_ERROR)));
+    return;
+  }
+
+  auto url_loader =
+      MakeGRPCLoader(request_url, MakeGetCompactBlocksParams(from, to));
+
+  UrlLoadersList::iterator it = url_loaders_list_.insert(
+      url_loaders_list_.begin(), std::move(url_loader));
+
+  StreamHandlersList::iterator handler_it = stream_handlers_list_.insert(
+      stream_handlers_list_.begin(),
+      std::make_unique<GetCompactBlocksGrpcStreamHandler>());
+
+  static_cast<GetCompactBlocksGrpcStreamHandler*>(handler_it->get())
+      ->set_callback(base::BindOnce(&ZCashRpc::OnGetCompactBlocksResponse,
+                                    weak_ptr_factory_.GetWeakPtr(),
+                                    std::move(callback), it, handler_it));
+
+  (*it)->DownloadAsStream(url_loader_factory_.get(), handler_it->get());
+}
+
+void ZCashRpc::OnGetCompactBlocksResponse(
+    ZCashRpc::GetCompactBlocksCallback callback,
+    UrlLoadersList::iterator it,
+    StreamHandlersList::iterator handler_it,
+    base::expected<std::vector<std::string>, std::string> result) {
+  url_loaders_list_.erase(it);
+  stream_handlers_list_.erase(handler_it);
+
+  if (!result.has_value()) {
+    std::move(callback).Run(
+        base::unexpected(l10n_util::GetStringUTF8(IDS_WALLET_INTERNAL_ERROR)));
+    return;
+  }
+
+  GetDecoder()->ParseCompactBlocks(
```

Comment:
We are parsing data in the background process

Reply:
Ok, since we're isolating this to a background process then it sounds like that would be a rule of 2 solution even though I don't think that was the intention for it. No concerns from me since that's the only reason I was thinking about it. FWIW, I vaguely remember GRPC being a bit different around handling of untrusted data relative to like JSON decoding from RPC calls, but that's the part I was trying to figure out. I'll take a look at the docs to get a refresher for myself, but in any case this wouldn't be a concern.